intelmq/bots/experts/sieve/README.md

```diff
@@ -0,0 +1,166 @@
+# Sieve Bot
+
+The sieve bot is used to filter and/or modify events based on a set of rules. The
+rules are specified in an external configuration file and with a syntax similar
+to the [Sieve language](http://sieve.info/) used for mail filtering.
+
+Each rule defines a set of matching conditions on received events. Events can be
+matched based on keys and values in the event. If the processed event matches a
+rule's conditions, the corresponding actions are performed. Actions can specify
+whether the event should be kept or dropped in the pipeline (filtering actions)
+or if keys and values should be changed (modification actions).
+
+
+## Examples
+
+The following excerpts illustrate some of the basic features of the sieve file
+format:
+
+```
+if :exists source.fqdn {
+  keep  // aborts processing of subsequent rules and forwards the event.
+}
+
+
+if :notexists source.abuse_contact || source.abuse_contact =~ '.*@example.com' {
+  drop  // aborts processing of subsequent rules and drops the event.
+}
+
+if source.ip << 192.0.0.0/24 {
+    add! comment = 'bogon'
+}
+
+if classification.type == ['phishing', 'malware'] && source.fqdn =~ '.*\.(ch|li)$' {
+  add! comment = 'domainabuse'
+  keep
+} elsif classification.type == 'scanner' {
+  add! comment = 'ignore'
+  drop
+} else {
+  remove comment
+}
+```
+
+
+## Parameters
+
+The sieve bot takes only one parameter:
+ * `file` - filesystem path the the sieve file
+
+
+## Reference
+
+### Sieve File Structure
+
+The sieve file contains an arbitrary number of rules of the form:
+
+```
+if EXPRESSION {
+    ACTIONS
+} elif EXPRESSION {
+    ACTIONS
+} else {
+    ACTIONS
+}
+```
+
+
+###  Expressions
+
+Each rule specifies on or more expressions to match an event based on its keys
+and values. Event keys are specified as strings without quotes. String values
+must be enclosed in single quotes. Numeric values can be specified as integers
+or floats and are unquoted. IP addresses and network ranges (IPv4 and IPv6) are
+specified with quotes. Following operators may be used to match events:
+
+ * `:exists` and `:notexists` match if a given key exists, for example:
+
+    ```if :exists source.fqdn { ... }```
+
+ * `==` and `!=` match for equality of strings and numbers, for example:
+
+   ```if feed.name != 'acme-security' || feed.accuracy == 100 { ... }```
+
+ * `:contains` matches on substrings.
```

Comment:
wish list: if we have contains, out of reasons of symmetry , may we have ``:notcontains`` ?